### ISSUE TITLE
WebGL Error when switching from a CubeGeometry to a EquirectGeometry

### ISSUE THREAD

**Issue by xXKeyleXx** [google/marzipano#171]
When you switch to a equirect panorama when you were on a cubemap panorama you get the following error:

Chrome:
  - marzipano.js:21 WebGL: INVALID_OPERATION: drawElements: no buffer is bound to enabled attribute

Firefox:
 - Error: WebGL warning: drawElements: Vertex attrib array 0 is enabled but has no buffer bound.

In the time of the transition you can see the equirect panorama but it disappears after that and throws the errors above.
If you swtich to the equirect panorama without beeing on a cubemap panorama first it works fine.

Marzipano version: v0.7.1

### Example:
http://dl.keyle.de/marzipano/

**Comment by tjgq** [comment]
Thanks for reporting!

I think I've nailed down the issue: we enable vertex attributes [when the shaders are created](https://github.com/google/marzipano/blob/master/src/renderers/WebGlCommon.js#L61), but never disable them. However, enabled attributes are a global setting, not per-shader. Since the equirect shader has a single active attribute but the cube shader has two, any equirect layer rendered after a cube layer will run with an enabled vertex attribute that isn't bound to anything. [This page](http://www.mjbshaw.com/2013/03/webgl-fixing-invalidoperation.html) explains the problem in some detail.

The fix I came up with is to enable the attributes when a layer is about to be rendered, and disable them once rendering is complete ([see here](https://github.com/tjgq/marzipano/commit/06b28627a979854e4bb266bb2459b39e0d396ca0)). Since this is potentially performance sensitive, I want to do some more testing before submitting. However, feel free to patch the fix and let me know if it solves the problem for you.
